lib/libc/stdio/printfcommon.h

```diff
@@ -279,82 +279,131 @@ __ujtoa(uintmax_t val, CHAR *endp, int base, int octzero, const char *xdigs)
 	return (cp);
 }
 
-#ifdef __CHERI_PURE_CAPABILITY__
-/*
- * Print the pointer details.  Format matches the kernel register dump format:
- * v:0 s:0 p:00000000 b:0000000000000000 l:0000000000000000 o:0 t:0
- *  v	tag (valid)
- *  s	sealed
- *  p	permissions
- *  b	base
- *  l	length
- *  o	offset
- *  t	type
+#if __has_feature(capabilities)
+/*-
+ * Print the pointer details.
+ * <address> [<permissions>,<base>-<top>] <attr>
+ *
+ * For null-derived capabilities, only the address is displayed.
+ *
+ * The address, base, and top are all printed in hex and honor
+ * requested precision by padding with leading zeroes.
+ *
+ * The permissions are zero or more of 'r' (LOAD), 'w' (STORE), 'x'
+ * (EXECUTE), 'R' (LOAD_CAP), and 'W' (STORE_CAP).
```

Comment:
Put `'x'` on the same line as `(EXECUTE)` or make it one line per permission?

Reply:
Fixed.  I was just using the default word-wrapping in Emacs (M-q), but I've just manually reflowed it.